packages/blade/src/components/Tabs/TabIndicator.tsx

```diff
@@ -0,0 +1,139 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+/* eslint-disable consistent-return */
+import React from 'react';
+import styled from 'styled-components';
+import { useTabsContext } from './TabsContext';
+import { castWebType, makeMotionTime, useTheme } from '~utils';
+import { useIsomorphicLayoutEffect } from '~utils/useIsomorphicLayoutEffect';
+import BaseBox from '~components/Box/BaseBox';
+
+const StyledTabIndicator = styled(BaseBox)(({ theme }) => {
+  return {
+    cursor: 'pointer',
+    '&:hover': {
+      backgroundColor: theme.colors.brand.primary[400],
+    },
+  };
+});
+
+const TabIndicator = ({
+  tabListContainerRef,
+}: {
+  tabListContainerRef: React.RefObject<HTMLElement | null>;
+}): React.ReactElement => {
+  const { theme } = useTheme();
+  const { selectedValue, baseId, variant } = useTabsContext();
+  const [hasMeasured, setHasMeasured] = React.useState(false);
+  const [activeElementDimensions, setActiveElementDimensions] = React.useState({
+    width: 0,
+    height: 0,
+    x: 0,
+    y: 0,
+  });
+
+  // Update the dimensions of the active element
+  const updateDimensions = React.useCallback(() => {
+    const tabItemId = `${baseId}-${selectedValue}-tabitem`;
+    const activeTabItem = document.getElementById(tabItemId);
+    if (!activeTabItem) return;
+
+    setActiveElementDimensions({
+      width: activeTabItem.offsetWidth,
+      height: activeTabItem.offsetHeight,
+      x: activeTabItem.offsetLeft,
+      y:
+        variant === 'filled'
+          ? // on filled variant the indicator is positioned on top of the tab item
+            // so no need to add offsetHeight
+            activeTabItem.offsetTop - 1
+          : activeTabItem.offsetTop + activeTabItem.offsetHeight - 1.5,
+    });
+  }, [baseId, selectedValue, tabListContainerRef, variant]);
+
+  // Update the dimensions when the selected value changes
+  useIsomorphicLayoutEffect(() => {
+    if (!selectedValue) return;
+
+    updateDimensions();
+
+    const id = requestAnimationFrame(() => {
+      setHasMeasured(true);
+    });
+
+    return () => {
+      if (id) {
+        cancelAnimationFrame(id);
+      }
+    };
+  }, [baseId, selectedValue]);
+
+  // Update the dimensions when the window resizes or when the font loads
+  React.useEffect(() => {
+    if (!tabListContainerRef.current) return;
+
+    // check for FontFace API support
+    // FontFaceAPI is widely supported but better to be safe than sorry
+    if ('fonts' in document) {
```

Comment:
why do we do this in tabs? we already have fallback fonts right?

Reply:
This is waiting for fonts to be fully loaded and recalculating that blue bottom indicator's dimensions, because on initial render the fonts are not yet loaded and that time the dimensions are different. 

Check the video: See how after the fonts are loaded the indicator adjusts it's width to accommodate the new width of the tab item: 

https://github.com/razorpay/blade/assets/35374649/093e8657-86dd-418a-a686-4207eef3b901